### PR TITLE
FOUR-9160: Create a Helper to get the all the categories

### PR DESCRIPTION
**Pull request by CarliPinell**
## Issue & Reproduction Steps
Create a Helper to get the all the categories

## Related Tickets & Packages
- FOUR-9160

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy

### CODE REVIEW

File: ProcessMaker/Helpers/ArrayHelper.php [hunk 1]
```diff
@@ -39,4 +40,35 @@ public static function getArrayDifferencesWithFormat(array $changedArray, array
...
+     * using the Tables information through Models
+     * The method returns a String value
+     * The packageName parameter is optional, only use with Packages. Default value Null
+     * @param string $modelName
+     * @param string $ids
+     * @param string $columnName
+     * @param string $packageName
+     * @return string
+     */
+    public static function getNamesByIds(string $modelName, string $ids, string $columnName, string $packageName = null): string
```
Comment (pmPaulis) [inline ProcessMaker/Helpers/ArrayHelper.php:55]: Please this ProcessMaker/Helpers/ArrayHelper.php is related to the function arrays I think so this function needs to transform array in string like 

```
public static function convertArrayToString($array, $concat = ', ')
{
       return 'A, B, C';
}
```

please, move the logic with queries into other class related to that

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: ProcessMaker/Helpers/ArrayHelper.php (hunk 1 above)
Comment (CarliPinell) [reply ProcessMaker/Helpers/ArrayHelper.php]: solved. A complete Refactor was done.

File: tests/unit/ProcessMaker/Models/ProcessCategoryTest.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+use Tests\TestCase;
+
+class ProcessCategoryTest extends TestCase
+{
+    /**
+     * Test ScriptCategory Model.
+     *
+     * @return void
+     */
+    public function testGetNamesByIds()
```
Comment (pmPaulis) [inline tests/unit/ProcessMaker/Models/ProcessCategoryTest.php:15]: Please in order to create more categories you can use:
```
$category = ProcessCategory::factory()->create();
$category->name;
        $category->id;
```

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: tests/unit/ProcessMaker/Models/ProcessCategoryTest.php (hunk 2 above)
Comment (CarliPinell) [reply tests/unit/ProcessMaker/Models/ProcessCategoryTest.php:15]: Thanks, suggestion taken and applied.

File: tests/unit/ProcessMaker/Models/ProcessCategoryTest.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        //Case 1: one Id
+        $category = ProcessCategory::factory()->create([
+            'name' => 'Screen Category 7',
+        ]);
+        $this->assertEquals(
+            $category->name,
+            ProcessCategory::getNamesByIds($category->id)
+        );
+
+        $category = ProcessCategory::factory()->create([
```
Comment (pmPaulis) [inline tests/unit/ProcessMaker/Models/ProcessCategoryTest.php:26]: Please you can use this example:
```
$category2 = ProcessCategory::factory()->create([
            'name' => 'Screen Category 33',
        ]);
        $this->assertEquals(
            $category->name. ', '$category2->name,
            ProcessCategory::getNamesByIds($category->id. '' . $category2->id)
        );
```

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: tests/unit/ProcessMaker/Models/ProcessCategoryTest.php (hunk 3 above)
Comment (CarliPinell) [reply tests/unit/ProcessMaker/Models/ProcessCategoryTest.php]: Good observation. Tahnks. Solved.

**Review by pmPaulis — APPROVED** [review]

**Pull request merged**